src/types/browse.d.ts

```diff
@@ -77,40 +78,52 @@ declare class Browse {
 /** *********
  * Browse results returned from server
  ********** */
-export interface BrowseResponse<ResponseType> extends Record<string, any> {
-  request?: Partial<BrowseRequestType>;
-  response?: Partial<ResponseType>;
-  result_id?: string;
+export interface BrowseResponse<ResponseType, OmittedRequestFields extends keyof BrowseRequestType>
+  extends Record<string, any> {
+
+  request: Omit<BrowseRequestType, OmittedRequestFields>;
```

Comment:
Yeah I know the documentation says the format of the request object should not be programmatically relied on.. but otoh you have declared a type for the fields here, so I improved them at least a little bit..